js/packages/teams-ai/src/AI.ts

```diff
@@ -359,19 +359,22 @@ export class AI<TState extends TurnState = TurnState> {
                 step_count = 0;
             }
 
+            // Add action handlers for tools function calling to temp state
+            state.temp.toolHandlers = this._actions as unknown as Map<string, actions.ActionEntry<TurnState>>;
```

Comment:
are we double casting? referencing the `as unknown as Map<string, actions.ActionEntry<TurnState>>;`

Reply:
Yeah I worked with Alex and this is how we got it to work in terms of casting to Map. The issue relates to how TurnState is kind of a circular dependency.